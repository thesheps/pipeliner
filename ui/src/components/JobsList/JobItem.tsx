import React from "react";
import { ListItem, ListItemText } from "@material-ui/core";
import { PlayArrow } from "@material-ui/icons/";

export interface JobItemProps {
  id: string;
  name: string;
}

export const JobItem = ({ id, name }: JobItemProps) => (
  <ListItem key={id} dense button>
    <PlayArrow></PlayArrow>
    <ListItemText id={id} primary={name} />
  </ListItem>
);
