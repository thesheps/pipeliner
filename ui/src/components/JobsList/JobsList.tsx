import React from "react";
import { List, makeStyles } from "@material-ui/core";

export interface Job {
  id: string;
  name: string;
}

export interface JobsListProps {
  jobs: Job[];
}

const useStyles = makeStyles(theme => ({
  root: {
    width: "100%",
    maxWidth: 360,
    backgroundColor: theme.palette.background.paper
  }
}));

export const JobsList = ({ jobs }: JobsListProps) => {
  const classes = useStyles();

  return (
    <List className={classes.root}>
      {jobs.map(j => (
        <div key={j.id}>{j.name}</div>
      ))}
    </List>
  );
};
