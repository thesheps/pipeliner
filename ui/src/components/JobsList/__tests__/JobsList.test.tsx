import React from "react";
import { render } from "@testing-library/react";

import { JobsList, Job } from "../JobsList";

describe("JobsList", () => {
  it("renders the specified Jobs", () => {
    const jobs: Job[] = [];

    const { container } = render(<JobsList jobs={jobs} />);
    expect(container).toMatchSnapshot();
  });
});
