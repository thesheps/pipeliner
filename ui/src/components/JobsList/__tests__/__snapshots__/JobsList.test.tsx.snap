// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobsList renders the specified Jobs 1`] = `
<div>
  <ul
    class="MuiList-root makeStyles-root-1 MuiList-padding"
  />
</div>
`;
